sklearn/preprocessing/tests/test_polynomial.py

```diff
@@ -322,18 +322,21 @@ def test_spline_transformer_periodic_splines_smoothness(degree):
     # numerical derivative is reasonably small (smaller than `tol` in absolute
     # value). We thus compute d-th numeric derivatives for d = 1, ..., `degree`
     # and compare them to `tol`.
-    # Note that the 0-th derivative is the function itself, such that
-    # we are also checking its continuity.
+    #
+    # Note that the 0-th derivative is the function itself, such that we are
+    # also checking its continuity.
     for d in range(1, degree + 1):
-        dXt = np.diff(dXt, axis=0) / delta  # Compute d-th numeric derivative
         # Check continuity of the (d-1)-th derivative
-        assert (np.abs(dXt) < tol).all()
+        diff = np.diff(dXt, axis=0)
+        assert np.abs(diff).max() < tol
```

Comment:
Note that this (and below) depends on `tol / delta ~ 10`. I.e. if we reduce (or increase) the number of rows in `X` substantially, this test will fail.

Reply:
I don't understand why though. Continuity should always be checked on the raw diff, assuming that the steps are fine grained enough, no?

Otherwise we "check continuity" by bounding the absolute value of the derivative itself which seems wrong to me.

Reply:
Given continuity, the "raw diff" scales with the order of the stepsize (`delta`). It will be roughly of the same order. Above, your stepsize is ~1e-3, so checking against `tol=1e-2` is fine. If we reduce the stepsize to 0.1, the diff will be of order `diff ~ 0.1 > tol  = 1e-2`.